internal/provider/deployment.go

```diff
@@ -40,7 +41,12 @@ type DeploymentData struct {
 
 func getDeploymentDeployer(d *schema.ResourceData, apiClient *apiClient) (DeploymentDeployer, error) {
```

Comment:
is that a factory function? if so please use `NewDeployer` and don't bother about being lowercase, it's already in the internal package and there's no need I believe to add `deployment` prefix, unless there're other deployers for things other than deployments?